data/parse_smogon_stats.py

```diff
@@ -31,7 +28,7 @@ def get_smogon_stats_file_name(game_mode, month_delta=1):
         game_mode = game_mode[:-5]
 
     # always use the `-0` file - the higher ladder is for noobs
-    smogon_url = "https://www.smogon.com/stats/{}-{}/moveset/{}-0.txt"
+    smogon_url = "https://www.smogon.com/stats/{}-{}/chaos/{}-0.json"
```

Comment:
hmm this is going to be a _massive_ file :sweat: I'll have to see how long this takes / how much memory it consumes

Reply:
As suspected, this is going to be a large file and will consume quite a bit of space.

An optimization I will do is modify the code to send in the Pokemon names of the opponent's team. This way the only data returned from this function can be the Pokemon that the opponent actually has. This code is only called in formats with team-preview (not randombattles, for examples) so this should be possible.

I'll leave that as a future optimization that I can take care of - not necessary in this PR.